extensions/runtimes/src/index.ts

```diff
@@ -308,4 +308,174 @@ export default async (composer: any): Promise<void> => {
+      // run dotnet install on the project
+      const { stderr: installError, stdout: installOutput } = await execAsync(`dotnet remove package ${packageName}`, {
+        cwd: path.join(runtimePath, 'azurewebapp'),
+      });
+      if (installError) {
+        throw new Error(installError);
+      }
+      return installOutput;
+    },
+    identifyManifest: (runtimePath: string, projName?: string): string => {
+      return path.join(runtimePath, `${projName}.csproj`);
+    },
+    run: async (project: any, localDisk: IFileStorage) => {
+      composer.log('RUN THIS C# PROJECT!');
+    },
+    buildDeploy: async (runtimePath: string, project: any, settings: any, profileName: string): Promise<string> => {
+      composer.log('BUILD FOR DEPLOY TO AZURE!');
+
+      let csproj = '';
+      // find publishing profile in list
+      const profile = project.settings.publishTargets.find((p) => p.name === profileName);
+      if (profile.type === 'azurePublish') {
+        csproj = 'Microsoft.BotFramework.Composer.WebApp.csproj';
+      } else if (profile.type === 'azureFunctionsPublish') {
+        csproj = 'Microsoft.BotFramework.Composer.Functions.csproj';
+      }
+      const publishFolder = path.join(runtimePath, 'bin', 'release', 'publishTarget');
+      const deployFilePath = path.join(runtimePath, '.deployment');
+      const dotnetProjectPath = path.join(runtimePath, csproj);
+
+      // Check for existing .deployment file, if missing, write it.
+      if (!(await fs.pathExists(deployFilePath))) {
+        const data = `[config]\nproject = ${csproj}`;
+
+        await fs.writeFile(deployFilePath, data);
+      }
+
+      // do the dotnet publish
+      try {
+        const configuration = JSON.parse(profile.configuration);
+        const runtimeIdentifier = configuration.runtimeIdentifier;
+
+        // Don't set self-contained and runtimeIdentifier for AzureFunctions.
+        let buildCommand = `dotnet publish "${dotnetProjectPath}" -c release -o "${publishFolder}" -v q`;
+
+        if (profile.type === 'azurePublish')
+        {
+          // if runtime identifier set, make dotnet runtime to self contained, default runtime identifier is win-x64, please refer to https://docs.microsoft.com/en-us/dotnet/core/rid-catalog
+          buildCommand = `dotnet publish "${dotnetProjectPath}" -c release -o "${publishFolder}" -v q --self-contained true -r ${runtimeIdentifier ?? 'win-x64'}`;
+        }
+        const { stdout, stderr } = await execAsync(
+          buildCommand,
+          {
+            cwd: runtimePath,
+          }
+        );
+        composer.log('OUTPUT FROM BUILD', stdout);
+        if (stderr) {
+          composer.log('ERR FROM BUILD: ', stderr);
+        }
+      } catch (err) {
+        composer.log('Error doing dotnet publish', err);
+        throw err;
+        return;
+      }
+      // Then, copy the declarative assets into the build artifacts folder.
```

Comment:
this is probably not necessary anymore with the new runtime structure

Reply:
It seems you have already commented this out, I will leave as is with commented out code